mmdet/apis/ote/extension/datasets/mmdataset.py

```diff
@@ -56,17 +61,24 @@ def get_annotation_mmdet_format(dataset_item: DatasetItemEntity, labels: List[La
         n = len(class_indices)
         gt_bboxes.extend([[box.x1 * width, box.y1 * height, box.x2 * width, box.y2 * height] for _ in range(n)])
         gt_labels.extend(class_indices)
+        if hasattr(annotation.shape, "points"):
```

Comment:
What kinds of shapes have "points" attribute? Only polygons? I'd consider replacing this with conversion of any shape to a polygon using `ShapeFactory` here, as it is done for bounding boxes.

Reply:
The benefit of checking in this fashion avoids adding a convert2mask param in the function arg and reduce the impact of code change in other places. I don't mind adding the parameter though. What do you think @Ilya-Krylov ?

Reply:
Segmentation mask can be represented by any valid shape, not necessarily polygon, so conversion should happen somewhere.

Reply:
And what about labels domain for counting? Is it the same as for detection?

Reply:
Agree with Pavel.

@eugene123tw Do you think that we need param in the function? What if we always return a dataset item containing some mask?

Reply:
@druzhkov-paul it's the same as detection and there's no clear way to know what task we are running here

Reply:
@druzhkov-paul @Ilya-Krylov it'd create some overhead for the detection task if we have many boxes per image to convert (e.g. >500 boxes) and I'm not sure about running unnecessary ops.

Reply:
properly is better than slowly

Reply:
sure. I can make the change. IMO if adding a param to avoid unnecessary operations to this function is not proper then I can make the change.

Reply:
@Ilya-Krylov updated

Reply:
Ok, let's go with this one for now.